src/analysis/Environments/Environments.ts

```diff
@@ -238,29 +206,37 @@ export function PauseButton({ cloudEnvironment, currentUser, pauseComputeAndRefr
     : null;
 }
 
-interface DecoratedResourceAttributes {
-  workspace: WorkspaceInfo;
-  unsupportedWorkspace: boolean;
+export interface UseWorkspacesResult {
+  workspaces: WorkspaceWrapper[];
+  refresh: () => Promise<void>;
+  loading: boolean;
 }
 
-type RuntimeWithWorkspace = DecoratedResourceAttributes & Runtime;
-type DiskWithWorkspace = DecoratedResourceAttributes & PersistentDisk;
-type AppWithWorkspace = DecoratedResourceAttributes & App;
+export type UseWorkspacesProvider = (
```

Comment:
```suggestion
export type UseWorkspacesState = (
```

Together with the previous comment about `UseWorkspacesResult`, since the Environments prop is named `useWorkspacesState`, can we keep the naming consistent across all the related types?

I still find "Provider" here misleading since it sounds like something that will provide "useWorkspaces", when it's actually the interface for "useWorkspaces" itself.

Reply:
how about UseWorkspacesStateProvider which returns UseWorkspacesState?

Reply:
I would interpret UseWorkspacesStateProvider which returns UseWorkspacesState as something that provides/returns a `useWorkspacesState` function.

I think both "use" and "provider" convey a simlilar meaning, so if they're both present in the name, it sounds like there's another layer involved.